packages/vite/src/node/ssr/__tests__/ssrTransform.spec.ts

```diff
@@ -211,8 +211,56 @@ test('should declare variable for imported super class', async () => {
   ).toMatchInlineSnapshot(`
     "const __vite_ssr_import_0__ = __vite_ssr_import__(\\"./dependency\\")
     const Foo = __vite_ssr_import_0__.Foo;
-    __vite_ssr_exports__.default = class A extends Foo {}
+    class A extends Foo {}
     class B extends Foo {}
+    Object.defineProperty(__vite_ssr_exports__, \\"default\\", { enumerable: true, value: A })
+    Object.defineProperty(__vite_ssr_exports__, \\"B\\", { enumerable: true, configurable: true, get(){ return B }})"
+  `)
+})
+
+// #4049
+test('should handle default export variants', async () => {
+  // default anonymous functions
+  expect(
+    (await ssrTransform(`export default function() {}\n`, null, null)).code
+  ).toMatchInlineSnapshot(`
+    "__vite_ssr_exports__.default = function() {}
+    "
+  `)
+  // default anonymous class
+  expect((await ssrTransform(`export default class {}\n`, null, null)).code)
+    .toMatchInlineSnapshot(`
+    "__vite_ssr_exports__.default = class {}
+    "
+  `)
```

Comment:
I'm not sure why, but stripping out the newline from the source + target caused `toMatchInlineSnapshot` to insist that the lines were not equivalent. There might be a bug with its handling of single-line code?

Reply:
Could apply `.trim()` to `code` before `toMatchInlineSnapshot`